### PR TITLE
Calculate Seq values the same way IPAM / Agent does

### PR DESCRIPTION
**Pull request by cgilmour**
So.. IPAM changed to make the tenant/segment sequence numbers 0-based when allocates IP addresses. And Agent decomposes the IP address to get the Tenant & Segment values to use in rule names.
This meant instead of producing rule names like `ROMANA-T1S1` they became `ROMANA-T0S0`.
But the k8s-listener is resolving tenant and segment names to produce a policy with ID numbers in it to send to k8s "agents". It would receive the tenant _id_ and the segment _seq_ values and produce a rule name `ROMANA-T2S3`. Since it failed to match a rule, it wouldn't inject parts for the network policy.

This makes the k8s-listener produce the same values that Agent would use when creating a rule name, so that when they're passed to the k8s "agents", they produce the same name, match and get the injected policy bits.

### CODE REVIEW

**Comment by cgilmour** [comment]
When the fix is reviewed/revised and finally merged, how do we want to release it all? Just bump this module as v0.8.0.1 or retag everything as v0.8.1?

**Comment by debedb** [comment]
My fault :). Question:
1. Is there a way to think about somehow catching this automatically? Should we make an issue for that?
2. Agent should really be re-querying other services, not caching results, it seems. Same question re issue.

**Comment by jbrendel** [comment]
Somehow seems to be wrong to have to decompose IP addresses to get tenant IDs. And I agree with Greg that the caching of those things doesn't seem to be ideal.

Caleb: Yes, 0.8.1 (not 0.8.0.1) and then re-release.

**Comment by cgilmour** [comment]
OK. Reviewed?

File: NetworkPolicy/k8s-listener.py [hunk 1]
```diff
@@ -566,20 +576,22 @@ def process(s, uid_by_ns, pd_by_uid):
...
+    logging.info("Discovered dst_segment_seq = %s" % dst_segment_seq)
 
     # That should be a romana policy object.
+    # We subtract 1 from the seq values to match IPAM/Agent behaviour in producing rule labels
     policy_definition = {
         "policy_name" : obj['object']['metadata']['name'],
         "policy_namespace" : obj['object']['metadata']['namespace'],
-        "owner_tenant_id" : tenant_id,
-        "target_segment_id" : dst_segment_id,
+        "owner_tenant_id" : tenant_seq - 1,
```
Comment (pritesh) [inline NetworkPolicy/k8s-listener.py:591]: going to throw TypeError here for None if #L392 ever comes true, same for other variables below, probably should be handled in core/pkg/../agent.py correctly in next release.

**Comment by pritesh** [comment]
rest looks good!

File: NetworkPolicy/k8s-listener.py (hunk 1 above)
Comment (cgilmour) [inline NetworkPolicy/k8s-listener.py:591]: For that circumstance, a TypeError is probably better than sending things that aren't correct. So I'll call that an accidental improvement. :-)

**Pull request merged**